infra/deploy.sh

```diff
@@ -682,6 +687,9 @@ startBanner
 checkRequiredTools
 populateParams $PARAMS_FILE
 
+# Check SKU availability and quotas
+validateSKUs $LOCATION $VALIDATE_SKUS_FLAG
```

Comment:
I moved the SKU validation check to a spot in the script before the resource group gets created. This will allow for a cleaner exit of the script without leaving artifacts behind in a customer's environment.